cnd/src/swap_protocols/rfc003/actions/bitcoin.rs

```diff
@@ -33,13 +33,16 @@ impl RefundAction<Bitcoin, BitcoinQuantity> for (Bitcoin, BitcoinQuantity) {
         htlc_params: HtlcParams<Bitcoin, BitcoinQuantity>,
         htlc_location: OutPoint,
         secret_source: &dyn SecretSource,
+        fund_transaction: &Transaction,
     ) -> Self::RefundActionOutput {
         let htlc = BitcoinHtlc::from(htlc_params.clone());
 
         SpendOutput {
             output: PrimedInput::new(
                 htlc_location,
-                htlc_params.asset,
+                BitcoinQuantity::from_satoshi(
+                    fund_transaction.output[htlc_location.vout as usize].value,
```

Comment:
❓ can this casting panic?

Reply:
should not. `vout` is a `u32` which fits into `usize` (which is either u32 on x86 or u64 on x86_64)